nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -275,11 +280,31 @@ BtNavigator::navigateToPose()
       geometry_msgs::msg::PoseStamped goal_pose;
       blackboard_->get("goal", goal_pose);
 
-      feedback_msg->distance_remaining = nav2_util::geometry_utils::euclidean_distance(
-        feedback_msg->current_pose.pose, goal_pose.pose);
+      // Get current path points
+      nav_msgs::msg::Path current_path;
+      blackboard_->get("path", current_path);
+
+      // Calculate distance on the path
+      double distance_remaining =
+        nav2_util::geometry_utils::calculate_path_length(current_path);
+
+      if (distance_remaining <= transform_tolerance_) {
+        estimated_navigation_time_remaining_ = rclcpp::Duration::from_seconds(0.0);
+      } else {
+        // Get current speed
+        current_linear_speed_ = odom_smoother_->getTwist().linear.x;
+
+        // Calculate estimated time taken to goal if speed is higher than 1mm/s
+        if (std::abs(current_linear_speed_) > 0.05) {
```

Comment:
This seems to be 5cm/s?

Reply:
Oops, I forgot to change that back! I meant for it to be 5mm/s :)